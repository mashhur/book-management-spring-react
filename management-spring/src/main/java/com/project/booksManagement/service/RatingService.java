package com.project.booksManagement.service;

import java.util.List;

import com.project.booksManagement.model.Ratings;

public interface RatingService {

	public Ratings addRatings(Ratings Customer) ;
	public Ratings updteRatings(Ratings Customer,Integer id);
	public void deleteCustomer(Integer id);
	public List<Ratings> viewRatings();
	public Ratings getRatings(Integer id);
}
