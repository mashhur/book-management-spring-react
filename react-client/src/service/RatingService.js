const axios = require("axios");

const BASE_URL = "http://localhost:8080/api/c/ratings";

class RatingsService {
	addRating(data) {
		return axios.post(`${BASE_URL}/add`, data);
	}
}

module.exports = new RatingsService();
