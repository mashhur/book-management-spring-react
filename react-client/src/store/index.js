import { configureStore } from "@reduxjs/toolkit";
import locationReducer from "./slices/locationSlice";
import userReducer from "./slices/userSlice";

export const store = configureStore({
	reducer: {
		location: locationReducer,
		user: userReducer,
	},
});
