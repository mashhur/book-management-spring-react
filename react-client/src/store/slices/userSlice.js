import { createSlice } from "@reduxjs/toolkit";

const inituser = JSON.parse(localStorage.getItem("user"));
const user = inituser || {
	cid: "",
	fullName: "",
	phoneNumber: "",
};
const initialState = {
	auth: localStorage.getItem("auth") || false,
	user: user,
	address: {
		aid: "",
		customerName: "",
		phoneNumber: "",
		address: "",
		mstate: "",
		pincode: "",
		customerId: "",
	},
	books: [],
	cart: [],
};

export const userSlice = createSlice({
	name: "user",
	initialState,
	reducers: {
		changeAuth: (state) => {
			state.auth = true;
		},
		setUser: (state, action) => {
			state.user = action.payload;
		},
		setAddress(state, action) {
			state.address = action.payload;
		},
		addBook(state, action) {
			state.books = [];
			state.books.push(action.payload);
		},
		addBooktoCart(state, action) {
			state.cart.push(action.payload);
		},
		addtoBuy(state) {
			const cart = state.cart;
			state.books = cart;
			state.cart = [];
		},
		clearOrderDetails(state) {
			state.books = [];
			state.address = {};
		},
		removeBook(state, action) {
			state.cart.splice(action.payload, 1);
		},
		clearAuth(state) {
			state.auth = false;
			state.user = {
				cid: "",
				fullName: "",
				phoneNumber: "",
			};
		},
	},
});

export const {
	changeAuth,
	setUser,
	setAddress,
	addBook,
	clearOrderDetails,
	removeBook,
	addBooktoCart,
	addtoBuy,
	clearAuth,
} = userSlice.actions;

export default userSlice.reducer;
