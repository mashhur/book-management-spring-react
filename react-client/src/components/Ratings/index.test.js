import React from "react";
import { mount, shallow } from "enzyme";
import RatingComponent from ".";
import {
	checkProp,
	findByTestAttr,
} from "../../tests/testUtils";
import { Provider } from "react-redux";
import { createTestStore } from "../../store";
import { setUser } from "../../store/slices/userSlice";
import RatingService from "../../service/RatingService";

const defaultProps = {
	book: {
		ratings: [],
		bid: 1,
	},
};

let store;
const setup = (store, props = {}) => {
	const setUpProps = { ...defaultProps, ...props };

	return mount(
		<Provider store={store}>
			<RatingComponent {...setUpProps} />
		</Provider>
	);
};

describe("Ratings component test casses", () => {
	beforeEach(() => {
		store = createTestStore();
	});

	it("renders without any error", async () => {
		const wrapper = setup(store);
		const component = await findByTestAttr(
			wrapper,
			"component-ratings"
		);
		expect(component.length).toBe(1);
	});

	describe("when ratings are not given", () => {
		it("should render the input field", async () => {
			const wrapper = setup(store);
			const component = await findByTestAttr(
				wrapper,
				"component-ratings-input"
			);
			expect(component.exists()).toBe(true);
		});
		it("should allow to change the ratings", async () => {
			store.dispatch(setUser({ cid: 25 }));

			const wrapper = setup(store);
			const component = await findByTestAttr(
				wrapper,
				"component-ratings-input"
			);
			expect(component.first().prop("disabled")).toBe(
				false
			);
		});
		it("can send the request", async () => {
			store.dispatch(setUser({ cid: 25 }));

			jest.spyOn(RatingService, "addRating");
			const wrapper = setup(store);
			const component = await findByTestAttr(
				wrapper,
				"component-ratings-input"
			);
			component.first().simulate("click");
			expect(RatingService.addRating).toHaveBeenCalledTimes(
				1
			);
		});
	});

	describe("when ratings are given", () => {
		it("should not render the input field", async () => {
			store.dispatch(setUser({ cid: 25 }));
			const props = {
				book: {
					ratings: [
						{
							mcid: "25",
							mbid: "18",
							rating: "5",
						},
					],
					bid: 18,
				},
			};

			const wrapper = setup(store, props);
			const component = await findByTestAttr(
				wrapper,
				"component-ratings-input"
			);
			expect(component.exists()).toBe(false);
		});
		it("cant send the request", async () => {
			store.dispatch(setUser({ cid: 25 }));
			const props = {
				book: {
					ratings: [
						{
							mcid: "25",
							mbid: "18",
							rating: "5",
						},
					],
					bid: 18,
				},
			};

			jest.spyOn(RatingService, "addRating");
			setup(store, props);
			expect(RatingService.addRating).toHaveBeenCalledTimes(
				0
			);
		});
		it("should not allow to change the ratings", async () => {
			store.dispatch(setUser({ cid: 25 }));
			const props = {
				book: {
					ratings: [
						{
							mcid: "25",
							mbid: "18",
							rating: "5",
						},
					],
					bid: 18,
				},
			};

			const wrapper = setup(store, props);
			const component = await findByTestAttr(
				wrapper,
				"component-ratings-input"
			);
			expect(component.exists()).toBe(false);
		});
	});

	it("should match the default snapshot", () => {
		const wrapper = setup(store);
		expect(wrapper).toMatchSnapshot();
	});

	it("doesnot throw warning with expected props", async () => {
		checkProp(RatingComponent, defaultProps);
	});
});
