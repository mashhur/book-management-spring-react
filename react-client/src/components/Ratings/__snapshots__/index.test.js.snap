// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings component test casses should match the default snapshot 1`] = `ReactWrapper {}`;
